ext/libxml/ruby_xml_node.c

```diff
@@ -45,13 +45,18 @@ static void rxml_node_deregisterNode(xmlNodePtr xnode)
   /* Has the node been wrapped and exposed to Ruby? */
   if (xnode->_private)
   {
+    /* Does it belong to another libxml user? */
+    if (!OWNED(xnode))
```

Comment:
In fact, you could optimise it in another way: set _private to the address of something belonging to libxml-ruby - for example, the hashtable. Then you can immediately determine if you own the node, and only do the hashtable lookup if you are actually going to access the object.